web/src/inbox_ui.ts

```diff
@@ -766,6 +826,27 @@ function get_sorted_row_dict<T extends DirectMessageContext | TopicContext>(
     return new Map([...row_dict].sort(([, a], [, b]) => b.latest_msg_id - a.latest_msg_id));
 }
 
+function sort_channel_folders(): void {
+    const sorted_channel_folders = [...channel_folders_dict.values()].sort((a, b) => {
+        // Sort OTHER_CHANNELS_FOLDER_ID last, then by name with PINNED_CHANNEL_FOLDER_ID first.
+        if (a.id === OTHER_CHANNELS_FOLDER_ID) {
```

Comment:
I feel like you might be able to just use a normal sorting function after mapping `OTHER_CHANNELS_FOLDER_ID` to `Infinity` for sorting, but maybe that has its own issues involving `Infinity-Infinity`. This seems fine.